### ISSUE TITLE
Feature Request: Add an option to hs.grid for utilizing screen:fullFrame()

### ISSUE THREAD

**Issue by elasticdog** [Hammerspoon/hammerspoon#792]
Currently, `hs.grid` hard-codes the use of `screen:frame()`; I'd love for that to be configurable so `screen:fullFrame()` could be used for its internal calculations instead.

My use case is that I auto-hide my dock and menu bar to maximize screen real estate...however, even with an auto-hiding dock, OS X reserves some pixels unnecessarily on whichever edge your dock is located (Googling reveals a lot historical complaints about this behavior). Always having that gap on the edge of the screen is not ideal, and utilizing `screen:fullFrame()` would fix the issue.

Let me know if you need any additional details, or a screenshot demonstrating the current behavior.

**Comment by cmsj** [comment]
@lowne any thoughts on this?

**Comment by lowne** [comment]
Addressed in #866 - this is how to use the new `frame` argument for your use case:
```lua
local scr=hs.screen.primaryScreen() -- or screen.find(whatever)
hs.grid.setGrid('5x3',scr,scr:fullFrame())
```
